File: MORM.Java/Classes/Comando.java
package MORM.Java.Classes;

public class Comando
{
    public String GetSelect(Class collectionItemClass, String where = "")
    {
        Tabela tabela = collectionItem.GetTabela();
        TList<Campo> campos = collectionItem.GetCampos();
        StringBuilder fieldsAtr = new StringBuilder();
        StringBuilder fields = new StringBuilder();

        for (Campo campo : campos)
        {
            AddString(fieldsAtr, campo.Atributo + " as \"" + campo.Atributo + "\"", ", ", "");
            AddString(fields, campo.Campo + " as " + campo.Atributo, ", ", "");
        }

        return
            "select " + fieldsAtr +
            " from (" +
              "select " + fields +
              " from " + vTabela.Nome + 
            ")" + (where != "" ? " where " + where : "")+
    }
    
    public String GetSelect(CollectionItem collectionItem, String where = "")
    {
        Tabela tabela = collectionItem.GetTabela();
        TList<Campo> campos = collectionItem.GetCampos();
        StringBuilder where = new StringBuilder();

        for (Campo campo : campos)
            if (campo.Tipo == CampoTipo.tfKey)
                AddString(where, campo.Atributo + " = " + GetValueStr(collectionItem, campo.Atributo), " and ", "");

        return GetSelect(collectionItem.Class, where);
    }
    
    public String GetInsert(CollectionItem collectionItem)
    {
        Tabela tabela = collectionItem.GetTabela();
        TList<Campo> campos = collectionItem.GetCampos();
        StringBuilder fields = new StringBuilder();
        StringBuilder values = new StringBuilder();
        
        for (Campo campo : campos)
        {
            AddString(fields, campo.Atributo, ", ", "");
            AddString(values, GetValueStr(collectionItem, campo.Atributo), ", ", "");        
        }

        return
            "insert into " + vTabela.Nome + 
            "(" + fields + 
            ") values (" + values + ")" ;
    }
    
    public String GetUpdate(CollectionItem collectionItem)
    {
        Tabela tabela = collectionItem.GetTabela();
        TList<Campo> campos = collectionItem.GetCampos();
        StringBuilder sets = new StringBuilder();
        StringBuilder where = new StringBuilder();

        for (Campo campo : campos)
            if (campo.Tipo == CampoTipo.tfKey)
                AddString(sets, campo.Atributo + " = " + GetValueStr(collectionItem, campo.Atributo), " and ", "");
            else
                AddString(sets, campo.Atributo + " = " + GetValueStr(collectionItem, campo.Atributo), ", ", "");

        return
            "update " + vTabela.Nome + 
            " set " + sets + 
            " where " + where ;
    }
    
    public String GetDelete(CollectionItem collectionItem)
    {
        Tabela tabela = collectionItem.GetTabela();
        TList<Campo> campos = collectionItem.GetCampos();
        StringBuilder where = new StringBuilder();
        
        for (Campo campo : campos)
            if (campo.Tipo == CampoTipo.tfKey)
                AddString(where, campo.Atributo + " = " + GetValueStr(collectionItem, campo.Atributo), " and ", "");
        
        return
            "delete from " + vTabela.Nome + 
            " where " + where;
    }
}

File: MORM.CSharp/MORM.Data/Classes/Comando.cs
using System;
using System.Linq;
using MORM.Util.Classes;

namespace MORM.Data.Classes
{
    public class Comando
    {
        public Comando(TipoDatabase tipoDatabase)
        {
            TipoDatabase = tipoDatabase;
        }
        
        //-- tipo database

        public TipoDatabase TipoDatabase { get; private set; }
        
        //-- tabela
        
        public Tabela GetTabela(Type type)        
        {
        	if (type.GetType().IsInstanceOfType(typeof(CollectionItem)))
            {
                var collection = Activator.CreateInstance(type) as CollectionItem;
                return collection.GetTabela();
            }
            else
                return null;
        }
        
        //-- campos

        public Campos GetCampos(Type type)
        {
        	if (type.GetType().IsInstanceOfType(typeof(CollectionItem)))
            {
                var collection = Activator.CreateInstance(type) as CollectionItem;
                return collection.GetCampos();
            }
            else
                return null;
        }

        //-- value

        public string GetValueStr(object obj, string atributo)
        {
            var value = obj.GetType().GetProperties()
            	.FirstOrDefault(x => x.Name == atributo)
            	.GetValue(obj);
            
            return (
                value == null ? "null" :
            	value is bool ? "'" + ((bool)value ? "T" : "F") + "'" :
            	value is DateTime ? TipoDatabase.GetValueData((DateTime)value) :
            	value is decimal ? value.ToString().Replace(",", ".") :
            	value is double ? value.ToString().Replace(",", ".") :
            	value is int ? value.ToString() :
            	value is string ? "'" + value.ToString().Replace("'", "''") + "'" : value.ToString());
        }

        private bool IsValueNull(object obj, string atributo)
        {
            var value = obj.GetType().GetProperties()
            	.FirstOrDefault(x => x.Name == atributo)
            	.GetValue(obj);
            
            return (
                value == null ? true :
            	value is bool ? (!((bool)value)) :
            	value is DateTime ? (((DateTime)value) == DateTime.MinValue) :
            	value is decimal ? (((decimal)value) == 0) :
            	value is double ? (((double)value) == 0) :
            	value is int ? (((int)value) == 0) :
            	value is string ? (value.ToString() == "") : true);
        }

        //-- string

        public void AddString(ref string str, string val, string sep, string ini = "")
        {
            str += (!string.IsNullOrWhiteSpace(str) ? sep : ini) + val;
        }
        
        //-- where

        public string GetWhereKey(object obj)
        {
            var campos = GetCampos(obj.GetType());
            var where = string.Empty;

            foreach (var campo in campos)
                if (campo.Tipo == CampoTipo.Key)
                    AddString(ref where, campo.Atributo + " = " + GetValueStr(obj, campo.Atributo), " and ");

            return where;
        }

        public string GetWhereAll(object obj)
        {
            var campos = GetCampos(obj.GetType());
            var where = string.Empty;

            foreach (var campo in campos)
                if (!IsValueNull(obj, campo.Atributo))
                    AddString(ref where, campo.Atributo + " = " + GetValueStr(obj, campo.Atributo), " and ");

            return where;
        }

        //-- select

        public string GetSelect(Type type, string where = "")
        {
            var tabela = GetTabela(type);
            var campos = GetCampos(type);
            var fieldsAtr = string.Empty;
            var fields = string.Empty;

            foreach (var campo in campos)
            {
                AddString(ref fieldsAtr, campo.Atributo + " as \"" + campo.Atributo + "\"", ", ");
                AddString(ref fields, campo.Nome + " as " + campo.Atributo, ", ");
            }

            return 
                "select " + fieldsAtr + 
            	" from (select " + fields + " from " + tabela.Nome + ")" +
                (!string.IsNullOrWhiteSpace(where) ? " where " + where : "") ;
        }

        public string GetSelect(object obj)
        {
            return GetSelect(obj.GetType(), GetWhereKey(obj));
        }

        //-- insert

        public string GetInsert(object obj)
        {
            var tabela = GetTabela(obj.GetType());
            var campos = GetCampos(obj.GetType());
            var fields = string.Empty;
            var values = string.Empty;

            foreach (var campo in campos)
            {
                AddString(ref fields, campo.Nome, ", ");
                AddString(ref values, GetValueStr(obj, campo.Atributo), ", ");
            }

            return 
                "insert into " + tabela.Nome + 
            	" (" + fields + 
            	") values (" + values + ")";
        }

        //-- update

        public string GetUpdate(object obj)
        {
            var tabela = GetTabela(obj.GetType());
            var campos = GetCampos(obj.GetType());
            var sets = string.Empty;
            var where = string.Empty;

            foreach (var campo in campos)
                if (campo.Tipo == CampoTipo.Key)
                    AddString(ref where, campo.Nome + " = " + GetValueStr(obj, campo.Atributo), " and ");
                else
                    AddString(ref sets, campo.Nome + " = " + GetValueStr(obj, campo.Atributo), ", ");

            return
                "update " + tabela.Nome + 
            	" set " + sets + 
            	" where " + where;
        }

        //-- delete

        public string GetDelete(object obj)
        {
            var tabela = GetTabela(obj.GetType());
            var campos = GetCampos(obj.GetType());
            var where = string.Empty;

            foreach (var campo in campos)
                if (campo.Tipo == CampoTipo.Key)
                    AddString(ref where, campo.Nome + " = " + GetValueStr(obj, campo.Atributo), " and ");

            return
                "delete from " + tabela.Nome + 
            	" where " + where;
        }
    }
}